src/shared/hwdb-util.c

```diff
@@ -650,14 +650,26 @@ int hwdb_update(const char *root, const char *hwdb_bin_dir, bool strict, bool co
         return r;
 }
 
-int hwdb_query(const char *modalias) {
+int hwdb_query(const char *modalias, const char *root) {
         _cleanup_(sd_hwdb_unrefp) sd_hwdb *hwdb = NULL;
-        const char *key, *value;
+        _cleanup_free_ char *hwdb_bin = NULL;
+        const char *key, *value, *p;
         int r;
 
         assert(modalias);
 
-        r = sd_hwdb_new(&hwdb);
+        if (!isempty(root))
+                NULSTR_FOREACH(p, hwdb_bin_paths) {
+                        hwdb_bin = path_join(root, p);
```

Comment:
this leaks memory on each iteration, the delcaration of hwdb_bin needs to be moved to this block

Reply:
Good catch, thanks. Fixed in the latest push.